# MAT281 - Laboratorio N°03



## Problema 01


<img src="https://imagenes.universia.net/gc/net/images/practicas-empleo/p/pr/pro/profesiones-con-el-avance-de-la-tecnologia.jpg" width="480" height="360" align="center"/>


EL conjunto de datos se denomina `ocupation.csv`, el cual contiene información de distintos usuarios (edad ,sexo, profesión, etc.).

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [385]:
import pandas as pd
import os

In [464]:
# cargar datos
df = pd.read_csv(os.path.join("ocupation.csv"), sep="|")
df.head(10)

user_id  age gender     occupation zip_code
0        1   24      M     technician    85711
1        2   53      F          other    94043
2        3   23      M         writer    32067
3        4   24      M     technician    43537
4        5   33      F          other    15213
5        6   42      M      executive    98101
6        7   57      M  administrator    91344
7        8   36      M  administrator    05201
8        9   29      M        student    01002
9       10   53      M         lawyer    90703

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:

### 1.- ¿Cuál es el número de observaciones en el conjunto de datos?

In [387]:
df['user_id'].max()

943

Por lo que vemos que hay 943 observaciones de personas, registrando cada una sus datos.

### 2.- ¿Cuál es el número de columnas en el conjunto de datos?

In [388]:
columnas = df.columns.values

In [389]:
len(columnas)

5

Son 5 columnas

### 3.- Imprime el nombre de todas las columnas

In [465]:
j=1
for i in columnas:
    print('el nombre de la columna', j, 'es', i)
    j+=1

el nombre de la columna 1 es user_id
el nombre de la columna 2 es age
el nombre de la columna 3 es gender
el nombre de la columna 4 es occupation
el nombre de la columna 5 es zip_code


### 4.- Imprima el índice del dataframe

In [466]:
a=df.index

In [467]:
for i in a:
    print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199
200
201
202
203
204
205
206
207
208
209
210
211
212
213
214
215
216
217
218
219
220
221
222
223
224
225
226
227
228
229
230
231
232
233
234
235
236
237
238
239
240
241
242
243
244
245
246
247
248
249
250
251
252
253
254
255
256
257
258
259
260
261
262
263
264
265
266
267
268
269
270
271
272
273
274
275
276
27

### 5.- ¿Cuál es el tipo de datos de cada columna?

In [393]:
df.dtypes

user_id        int64
age            int64
gender        object
occupation    object
zip_code      object
dtype: object

### 6.- Describir el conjunto de datos (**hint**: .describe())

In [394]:
df.describe()

user_id         age
count  943.000000  943.000000
mean   472.000000   34.051962
std    272.364951   12.192740
min      1.000000    7.000000
25%    236.500000   25.000000
50%    472.000000   31.000000
75%    707.500000   43.000000
max    943.000000   73.000000

### 7.- Imprimir solo la columna de **occupation**.

In [395]:
df['occupation']

0         technician
1              other
2             writer
3         technician
4              other
           ...      
938          student
939    administrator
940          student
941        librarian
942          student
Name: occupation, Length: 943, dtype: object

### 8.- ¿Cuántas ocupaciones diferentes hay en este conjunto de datos?

podemos ocupar la libreria colecctions

In [396]:
from collections import Counter

ahora usamos la funcion $\textit{Counter}$, que nos dará un diccionario donde la llave es el oficio y el valor su frecuencia

In [397]:
Counter(df['occupation'])

Counter({'technician': 27,
         'other': 105,
         'writer': 45,
         'executive': 32,
         'administrator': 79,
         'student': 196,
         'lawyer': 12,
         'educator': 95,
         'scientist': 31,
         'entertainment': 18,
         'programmer': 66,
         'librarian': 51,
         'homemaker': 7,
         'artist': 28,
         'engineer': 67,
         'marketing': 26,
         'none': 9,
         'healthcare': 16,
         'retired': 14,
         'salesman': 12,
         'doctor': 7})

ahora veamos solo los oficios

In [398]:
Counter(df['occupation']).keys()

dict_keys(['technician', 'other', 'writer', 'executive', 'administrator', 'student', 'lawyer', 'educator', 'scientist', 'entertainment', 'programmer', 'librarian', 'homemaker', 'artist', 'engineer', 'marketing', 'none', 'healthcare', 'retired', 'salesman', 'doctor'])

finalmente, veamos la cantidad de llaves

In [399]:
len(Counter(df['occupation']).keys())

21

Por lo que hay 21 oficios distintos

### 9.- ¿Cuál es la ocupación más frecuente?

In [400]:
Counter(df['occupation']).values()

dict_values([27, 105, 45, 32, 79, 196, 12, 95, 31, 18, 66, 51, 7, 28, 67, 26, 9, 16, 14, 12, 7])

In [401]:
lista=[]
for i in Counter(df['occupation']).values():
    lista.append(i)

In [402]:
lista.sort()
lista

[7,
 7,
 9,
 12,
 12,
 14,
 16,
 18,
 26,
 27,
 28,
 31,
 32,
 45,
 51,
 66,
 67,
 79,
 95,
 105,
 196]

vemos que el empleo más frecuente es el que aparece 196 veces, lo que según nuestro diccionario es 'student'

### 10.- ¿Cuál es la edad media de los usuarios?

In [403]:
print('la edad promedio es ' ,df['age'].mean(), 'años')

la edad promedio es  34.05196182396607 años


### 11.- ¿Cuál es la edad con menos ocurrencia?

procedemos análogamente a la pregunta 10

In [404]:
lista2=[]
for i in Counter(df['age']).values():
    lista2.append(i)
lista2.sort()
lista2

[1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 4,
 5,
 5,
 6,
 6,
 6,
 9,
 9,
 10,
 11,
 12,
 12,
 13,
 14,
 14,
 15,
 17,
 17,
 18,
 19,
 19,
 20,
 20,
 20,
 21,
 21,
 21,
 22,
 23,
 23,
 25,
 26,
 27,
 27,
 28,
 28,
 32,
 32,
 33,
 34,
 35,
 36,
 37,
 38,
 39]

buscamos la edad que aparecen 1 vez

In [405]:
 Counter(df['age'])

Counter({24: 33,
         53: 12,
         23: 28,
         33: 26,
         42: 21,
         57: 9,
         36: 21,
         29: 32,
         39: 22,
         28: 36,
         47: 14,
         45: 15,
         49: 19,
         21: 27,
         30: 39,
         35: 27,
         40: 21,
         26: 34,
         25: 38,
         32: 28,
         41: 10,
         7: 1,
         38: 17,
         20: 32,
         19: 23,
         27: 35,
         18: 18,
         22: 37,
         37: 19,
         16: 5,
         50: 20,
         31: 25,
         51: 20,
         17: 14,
         48: 20,
         34: 17,
         43: 13,
         60: 9,
         55: 11,
         15: 6,
         61: 3,
         44: 23,
         54: 4,
         59: 3,
         46: 12,
         13: 5,
         52: 6,
         56: 6,
         14: 3,
         66: 1,
         62: 2,
         11: 1,
         65: 3,
         68: 2,
         63: 3,
         64: 2,
         10: 1,
         73: 1,
         58: 3,
         69: 2,
    

que serían 7,66,11,10,73, que aparecen sólo una vez

In [457]:
df['age'][10]

39

### 12.- Encontrar la edad promedio según la variable **occupation**

In [462]:
trabajos=[]
for j in df['occupation']:
    suma=0
    contador=0
    for i in range(len(df['occupation'])):
        if df['occupation'][i]==j:
            suma+=df['age'][i]
            contador+=1
    if j not in trabajos:
        trabajos.append(j)
        print('el promedio de edad de ', j, 'es de', int(suma/(contador+1)), 'años aproximadamente')

el promedio de edad de  technician es de 31 años aproximadamente
el promedio de edad de  other es de 34 años aproximadamente
el promedio de edad de  writer es de 35 años aproximadamente
el promedio de edad de  executive es de 37 años aproximadamente
el promedio de edad de  administrator es de 38 años aproximadamente
el promedio de edad de  student es de 21 años aproximadamente
el promedio de edad de  lawyer es de 33 años aproximadamente
el promedio de edad de  educator es de 41 años aproximadamente
el promedio de edad de  scientist es de 34 años aproximadamente
el promedio de edad de  entertainment es de 27 años aproximadamente
el promedio de edad de  programmer es de 32 años aproximadamente
el promedio de edad de  librarian es de 39 años aproximadamente
el promedio de edad de  homemaker es de 28 años aproximadamente
el promedio de edad de  artist es de 30 años aproximadamente
el promedio de edad de  engineer es de 35 años aproximadamente
el promedio de edad de  marketing es de 36 años

## Problema 02


<img src="https://image.freepik.com/vector-gratis/varios-automoviles-dibujos-animados_23-2147613095.jpg" width="360" height="360" align="center"/>


EL conjunto de datos se denomina `Automobile_data.csv`, el cual contiene información tal como: compañia, precio, kilometraje, etc.

Lo primero es cargar el conjunto de datos y ver las primeras filas que lo componen:

In [496]:
# cargar datos
df = pd.read_csv(os.path.join("Automobile_data.csv")).set_index('index')
df.head(10)

company   body-style  wheel-base  length engine-type  \
index                                                             
0      alfa-romero  convertible        88.6   168.8        dohc   
1      alfa-romero  convertible        88.6   168.8        dohc   
2      alfa-romero    hatchback        94.5   171.2        ohcv   
3             audi        sedan        99.8   176.6         ohc   
4             audi        sedan        99.4   176.6         ohc   
5             audi        sedan        99.8   177.3         ohc   
6             audi        wagon       105.8   192.7         ohc   
9              bmw        sedan       101.2   176.8         ohc   
10             bmw        sedan       101.2   176.8         ohc   
11             bmw        sedan       101.2   176.8         ohc   

      num-of-cylinders  horsepower  average-mileage    price  
index                                                         
0                 four         111               21  13495.0  
1                 four         111               21  16500.0  
2                  six         154               19  16500.0  
3                 four         102               24  13950.0  
4                 five         115               18  17450.0  
5                 five         110               19  15250.0  
6                 five         110               19  18920.0  
9                 four         101               23  16430.0  
10                four         101               23  16925.0  
11                 six         121               21  20970.0

El objetivo es tratar de obtener la mayor información posible de este conjunto de datos. Para cumplir este objetivo debe resolver las siguientes problemáticas:

### 1. Elimine los valores nulos (Nan)

veamos cuantos valores faltantes hay

In [497]:
df.isna().sum()

company             0
body-style          0
wheel-base          0
length              0
engine-type         0
num-of-cylinders    0
horsepower          0
average-mileage     0
price               3
dtype: int64

eliminamos las filas con datos faltantes

In [498]:
df = df[df['company'].notna()]
df = df[df['body-style'].notna()]
df = df[df['wheel-base'].notna()]
df = df[df['length'].notna()]
df = df[df['engine-type'].notna()]
df = df[df['num-of-cylinders'].notna()]
df = df[df['horsepower'].notna()]

df = df[df['average-mileage'].notna()]
df = df[df['price'].notna()]

In [499]:
df.isna().sum()

company             0
body-style          0
wheel-base          0
length              0
engine-type         0
num-of-cylinders    0
horsepower          0
average-mileage     0
price               0
dtype: int64

vemos que ya no hay valores faltantes

### Arreglemos el indice

vemos que el indice no está bien, se salta números, así que lo arreglaremos

In [500]:
index=[]

for i in range(len(df['company'])):
    index.append(i)
    
df['indice']=index
df.head(10)

company   body-style  wheel-base  length engine-type  \
index                                                             
0      alfa-romero  convertible        88.6   168.8        dohc   
1      alfa-romero  convertible        88.6   168.8        dohc   
2      alfa-romero    hatchback        94.5   171.2        ohcv   
3             audi        sedan        99.8   176.6         ohc   
4             audi        sedan        99.4   176.6         ohc   
5             audi        sedan        99.8   177.3         ohc   
6             audi        wagon       105.8   192.7         ohc   
9              bmw        sedan       101.2   176.8         ohc   
10             bmw        sedan       101.2   176.8         ohc   
11             bmw        sedan       101.2   176.8         ohc   

      num-of-cylinders  horsepower  average-mileage    price  indice  
index                                                                 
0                 four         111               21  13495.0       0  
1                 four         111               21  16500.0       1  
2                  six         154               19  16500.0       2  
3                 four         102               24  13950.0       3  
4                 five         115               18  17450.0       4  
5                 five         110               19  15250.0       5  
6                 five         110               19  18920.0       6  
9                 four         101               23  16430.0       7  
10                four         101               23  16925.0       8  
11                 six         121               21  20970.0       9

In [501]:
df.set_index("indice", inplace=True)

In [502]:
df.head(10)

company   body-style  wheel-base  length engine-type  \
indice                                                             
0       alfa-romero  convertible        88.6   168.8        dohc   
1       alfa-romero  convertible        88.6   168.8        dohc   
2       alfa-romero    hatchback        94.5   171.2        ohcv   
3              audi        sedan        99.8   176.6         ohc   
4              audi        sedan        99.4   176.6         ohc   
5              audi        sedan        99.8   177.3         ohc   
6              audi        wagon       105.8   192.7         ohc   
7               bmw        sedan       101.2   176.8         ohc   
8               bmw        sedan       101.2   176.8         ohc   
9               bmw        sedan       101.2   176.8         ohc   

       num-of-cylinders  horsepower  average-mileage    price  
indice                                                         
0                  four         111               21  13495.0  
1                  four         111               21  16500.0  
2                   six         154               19  16500.0  
3                  four         102               24  13950.0  
4                  five         115               18  17450.0  
5                  five         110               19  15250.0  
6                  five         110               19  18920.0  
7                  four         101               23  16430.0  
8                  four         101               23  16925.0  
9                   six         121               21  20970.0

Ahora sí seguimos

### 2. Encuentra el nombre de la compañía de automóviles más cara

In [503]:
s=df['company'][0]
promedios=[]


suma=0
contador=0
for i in range(len(df['company'])-1):
    if df['company'][i]==df['company'][i+1]:
        suma+=df['price'][i]
        contador+=1
    else:
        if i!=2:
            print('la compañia', s, 'tiene de promedio $', suma/(contador+1) )
            promedios.append(suma/(contador+1) )
            s=df['company'][i]
            contador=0
            a=suma/(contador+1)
            t=True
            for j in promedios:
                if t==True:
                    if a>=j:
                        t=True
                    else:
                        t=False
            max=a
            caro=df['company'][i]
    
print('y el más caro en promedio es ', caro, 'que cuesta $', max, 'en promedio')   

la compañia alfa-romero tiene de promedio $ 12774.166666666666
la compañia audi tiene de promedio $ 33840.833333333336
la compañia bmw tiene de promedio $ 71497.0
la compañia chevrolet tiene de promedio $ 110434.0
la compañia dodge tiene de promedio $ 80369.33333333333
la compañia honda tiene de promedio $ 241108.0
la compañia isuzu tiene de promedio $ 102969.33333333333
la compañia jaguar tiene de promedio $ 67767.6
la compañia mazda tiene de promedio $ 108399.5
la compañia mercedes-benz tiene de promedio $ 113041.25
la compañia mitsubishi tiene de promedio $ 96022.2
la compañia nissan tiene de promedio $ 257069.5
la compañia porsche tiene de promedio $ 79415.28571428571
la compañia toyota tiene de promedio $ 144913.0
y el más caro en promedio es  volkswagen que cuesta $ 579652.0 en promedio


### 3. Imprimir todos los detalles de Toyota Cars

In [504]:
import pandas as pd

In [505]:

tbody_style=[]
twhee_base=[]
tlength=[]
tengine_typ=[]
tnum_of_cylinders=[]
thorsepower=[]
taverage_mileage=[]
tprice=[]

In [517]:
tlength

In [522]:
s='toyota'

for i in range(len(df['company'])):
    if df['company'][i]==s:
        tbody_style.append(df['body-style'][i])
        twhee_base.append(df['wheel-base'][i])
        tlength=[].append(df['length'][i])
        tengine_typ=[].append(df['engine-type'][i])
        tnum_of_cylinders=[].append(df['num-of-cylinders'][i])
        thorsepower=[].append(df['horsepower'][i])
        taverage_mileage=[].append(df['average-mileage'][i])
        tprice=[].append(df['price'][i])

In [519]:
Toyota=pd.DataFrame()

In [520]:
Toyota['body-style']=tbody_style
Toyota['wheel-base']=twhee_base
Toyota['length']=tlength
Toyota['engine-type']=tengine_typ
Toyota['num-of-cylinders']=tnum_of_cylinders
Toyota['horsepower']=thorsepower
Toyota['average-mileage']=taverage_mileage
Toyota['price']=tprice


In [521]:
Toyota.describe()

wheel-base
count    20.00000
mean     96.90000
std       2.67857
min      95.70000
25%      95.70000
50%      95.70000
75%      97.30000
max     104.50000

### 4. Cuente el total de automóviles por compañía

In [477]:
s=df['company'][0]
maximos=[]

for i in range(len(df['company'])-1):
    maximos.append(0)
    
for i in range(len(df['company'])-1):
    if df['price'][i]>maximos[i]:
            maximos[i]=df['price'][i]
    if df['company'][i]==df['company'][i+1]:
        j=1
    else:
        if i!=2:
            print('la compañia', s, 'tiene', i, 'autos, y su auto más caro vale', maximos[i-1] )
            s=df['company'][i]
        

la compañia alfa-romero tiene 6 autos, y su auto más caro vale 15250.0
la compañia audi tiene 12 autos, y su auto más caro vale 41315.0
la compañia bmw tiene 15 autos, y su auto más caro vale 6295.0
la compañia chevrolet tiene 17 autos, y su auto más caro vale 6377.0
la compañia dodge tiene 20 autos, y su auto más caro vale 12945.0
la compañia honda tiene 21 autos, y su auto más caro vale 10345.0
la compañia isuzu tiene 24 autos, y su auto más caro vale 35550.0
la compañia jaguar tiene 29 autos, y su auto más caro vale 11845.0
la compañia mazda tiene 33 autos, y su auto más caro vale 40960.0
la compañia mercedes-benz tiene 37 autos, y su auto más caro vale 6989.0
la compañia mitsubishi tiene 42 autos, y su auto más caro vale 7349.0
la compañia nissan tiene 44 autos, y su auto más caro vale 34028.0
la compañia porsche tiene 51 autos, y su auto más caro vale 8778.0
la compañia toyota tiene 55 autos, y su auto más caro vale 7995.0


### 5. Encuentra el coche con el precio más alto por compañía

Está desarrollado en el ejercicio anterior

### 6. Encuentre el kilometraje promedio (**average-mileage**) de cada compañía automotriz

In [421]:
d=df['company'][0]
promedios=[]


suma=0
contador=0
for i in range(len(df['company'])-1):
    if df['company'][i]==df['company'][i+1]:
        contador+=1
        suma+=df['average-mileage'][i]
        j=1
    else:
        if i!=2:
            print('la compañia', d, 'tiene kilometraje promedio', suma/(contador+1) )
            d=df['company'][i]
            suma=0
            contador=0

la compañia alfa-romero tiene kilometraje promedio 17.166666666666668
la compañia audi tiene kilometraje promedio 16.5
la compañia bmw tiene kilometraje promedio 28.333333333333332
la compañia chevrolet tiene kilometraje promedio 15.5
la compañia dodge tiene kilometraje promedio 18.0
la compañia honda tiene kilometraje promedio 0.0
la compañia isuzu tiene kilometraje promedio 10.0
la compañia jaguar tiene kilometraje promedio 21.8
la compañia mazda tiene kilometraje promedio 14.5
la compañia mercedes-benz tiene kilometraje promedio 23.25
la compañia mitsubishi tiene kilometraje promedio 27.6
la compañia nissan tiene kilometraje promedio 8.5
la compañia porsche tiene kilometraje promedio 26.0
la compañia toyota tiene kilometraje promedio 25.25


### 7. Ordenar todos los autos por columna de precio (**price**)

In [478]:
df.set_index("price", inplace=True)

In [479]:
df.sort_values(by="price")
df.head(10)

company   body-style  wheel-base  length engine-type  \
price                                                               
13495.0  alfa-romero  convertible        88.6   168.8        dohc   
16500.0  alfa-romero  convertible        88.6   168.8        dohc   
16500.0  alfa-romero    hatchback        94.5   171.2        ohcv   
13950.0         audi        sedan        99.8   176.6         ohc   
17450.0         audi        sedan        99.4   176.6         ohc   
15250.0         audi        sedan        99.8   177.3         ohc   
18920.0         audi        wagon       105.8   192.7         ohc   
16430.0          bmw        sedan       101.2   176.8         ohc   
16925.0          bmw        sedan       101.2   176.8         ohc   
20970.0          bmw        sedan       101.2   176.8         ohc   

        num-of-cylinders  horsepower  average-mileage  
price                                                  
13495.0             four         111               21  
16500.0             four         111               21  
16500.0              six         154               19  
13950.0             four         102               24  
17450.0             five         115               18  
15250.0             five         110               19  
18920.0             five         110               19  
16430.0             four         101               23  
16925.0             four         101               23  
20970.0              six         121               21

## Problema 03


Siguiendo la temática de los automóviles, resuelva los siguientes problemas:

### a) Subproblema 01
A partir de los siguientes diccionarios:

In [480]:
GermanCars = {'Company': ['Ford', 'Mercedes', 'BMV', 'Audi'],
              'Price': [23845, 171995, 135925 , 71400]}
japaneseCars = {'Company': ['Toyota', 'Honda', 'Nissan', 'Mitsubishi '], 
                'Price': [29995, 23600, 61500 , 58900]}

* Cree dos dataframes (**carsDf1** y **carsDf2**) según corresponda.
* Concatene ambos dataframes (**carsDf**) y añada una llave ["Germany", "Japan"], según corresponda.

In [481]:
carsDf1=pd.DataFrame()

In [482]:
carsDf1['Company']=['Ford', 'Mercedes', 'BMV', 'Audi']
carsDf1['Price']=[23845, 171995, 135925 , 71400]


In [483]:
carsDf1

Company   Price
0      Ford   23845
1  Mercedes  171995
2       BMV  135925
3      Audi   71400

Ahora creamos el segundo DataFrame

In [484]:
carsDf2=pd.DataFrame()

In [485]:
carsDf2['Company']=['Toyota', 'Honda', 'Nissan', 'Mitsubishi ']
carsDf2['Price']=[29995, 23600, 61500 , 58900]

In [486]:
carsDf2

Company  Price
0       Toyota  29995
1        Honda  23600
2       Nissan  61500
3  Mitsubishi   58900

In [487]:
diccionario={'Germany':carsDf1, 'Japan': carsDf2 }

In [488]:
diccionario

{'Germany':     Company   Price
 0      Ford   23845
 1  Mercedes  171995
 2       BMV  135925
 3      Audi   71400,
 'Japan':        Company  Price
 0       Toyota  29995
 1        Honda  23600
 2       Nissan  61500
 3  Mitsubishi   58900}

### b) Subproblema 02

A partir de los siguientes diccionarios:

In [489]:
Car_Price = {'Company': ['Toyota', 'Honda', 'BMV', 'Audi'], 'Price': [23845, 17995, 135925 , 71400]}
car_Horsepower = {'Company': ['Toyota', 'Honda', 'BMV', 'Audi'], 'horsepower': [141, 80, 182 , 160]}

* Cree dos dataframes (**carsDf1** y **carsDf2**) según corresponda.
*  Junte ambos dataframes (**carsDf**) por la llave **Company**.

In [490]:
carsDf1=pd.DataFrame()

In [491]:
carsDf1['Company']=['Toyota', 'Honda', 'BMV', 'Audi']
carsDf1['Price']=[23845, 17995, 135925 , 71400]

In [492]:
carsDf2=pd.DataFrame()

In [493]:
carsDf2['Company']=['Toyota', 'Honda', 'BMV', 'Audi']
carsDf2['Price']=[141, 80, 182 , 160]

In [494]:
carsDf={'Company':[carsDf2,carsDf1]}

In [495]:
carsDf

{'Company': [  Company  Price
  0  Toyota    141
  1   Honda     80
  2     BMV    182
  3    Audi    160,
    Company   Price
  0  Toyota   23845
  1   Honda   17995
  2     BMV  135925
  3    Audi   71400]}